core/src/main/java/org/bitcoinj/core/Wallet.java

```diff
@@ -1817,6 +1842,9 @@ private void receive(Transaction tx, StoredBlock block, BlockChain.NewBlockType
             log.info("  <-pending");
 
         if (bestChain) {
+            boolean wasDead = dead.remove(txHash) != null;
```

Comment:
Right, thanks for the explanation. It's been too long since I looked at this code and I'm forgetting the details of how it works.